### PR TITLE
Presign URL support

### PR DESCRIPTION
**Pull request by sandeep-paliwal**
Added support to share private files using presign URLs
All file URLs will now be routed via CDN so made changes to getURL to return appropriate URLs

## How Has This Been Tested?
unit test, manual tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [x ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x ] My code follows the code style of this project.
- [x ] My change requires a change to the documentation.
- [ x] I have updated the documentation accordingly.
- [ x] I have read the **CONTRIBUTING** document.
- [ x] I have added tests to cover my changes.
- [x ] All new and existing tests passed.

### CODE REVIEW

File: lib/Files.js [hunk 1]
```diff
@@ -329,6 +329,21 @@ class Files {
     throwNotImplemented('_getUrl')
   }
 
+  /**
+   * [INTERNAL]
+   *
+   * @param {RemotePathString} filePath {@link RemotePathString}
+   * @param {object} options Options for presign URL
```
Comment (shazron) [inline lib/Files.js:323]: options not documented?

File: lib/Files.js [hunk 2]
```diff
@@ -719,6 +734,23 @@ class Files {
     await _copyFiles(srcToDestFilesMapping, options)
     return srcToDestFilesMapping
   }
+
+  /**
+   * Generate pre-sign URLs for a private file
+   *
+   * @param {RemotePathString} filePath {@link RemotePathString}
+   * @param {object} options Options for presign URL
```
Comment (shazron) [inline lib/Files.js:729]: options not documented?

File: lib/impl/AzureBlobFiles.js [hunk 3]
```diff
@@ -329,7 +334,23 @@ class AzureBlobFiles extends Files {
...
+    return 'https://' + DEFAULT_CDN_STORAGE_HOST + azureURL.substring(index + AZURE_STORAGE_DOMAIN.length)
+  }
+
+  /**
+   * @memberof AzureBlobFiles
+   * @override
+   * @private
+   */
+  async _getPresignUrl (filePath, options) {
+    if (!options || !options.expiryInSeconds) {
```
Comment (shazron) [inline lib/impl/AzureBlobFiles.js:348]: options.blobName is also required

Comment (sandeep-paliwal) [reply lib/impl/AzureBlobFiles.js:348]: blobName is not passed by users in options but as the filePath argument. We set blobName in next line as an option for TVM. So we won't need to check blobName requirement here.

**Comment by moritzraho** [comment]
lgtm

**Pull request merged**